source/perflib/basename.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifdef	_WIN32
+
+#include <string.h>
+
+/*
+ * windows variant of basename(3). works on ASCIIZ only.
+ * simple and perhaps naive implementation too.
+ */
+const char * basename(const char *path)
+{
+	const char *rv;
+
+	rv = (const char *)strrchr(path, '\\');
```

Comment:
Both `\\` and `/` should be supported as directory separator.

Reply:
The updated code supports directory separators of more than one character, but both `\\` and `/` are only single characters. Do we need support for > 1 character directory separators?

> windows only so we can rely on directory separator to be a backslash.

Windows APIs support `/` as a directory separator, so we should too.

However, supporting `\\` and `/`  (and only those two) might be better - again, you can pass something like `C:\Users\Tom/projects/openssl` to Windows APIs and it will work as you would expect

Reply:
thanks for clarification. it makes sense now. I did not know about '/' on windows.